### ISSUE TITLE
npm install error cloning into bare repository

### ISSUE THREAD

**Issue by walhow** [npm/npm#15080]
#### I'm opening this issue because:

  - [x ] npm is doing something I don't understand.

#### What's going wrong?
Using npm install, can't install a package from private Stash repository. Note that other packages from same repository can be installed. Getting error: 

`npm ERR! git clone --template=C:\Users\user-xyz\AppData\Roaming\npm-cache\_git-remotes\_templates --mirror https://acme.com/scm/cf/cf-preferences-js.git C:\Users\user-xyz\AppData\Roaming\npm-cache\_git-remotes\git-https-acme.com-scm-cf-cf-preferences-js-git-90432a4a: Cloning into bare repository 'C:\Users\user-xyz\AppData\Roaming\npm-cache\_git-remotes\git-https-acme.com-scm-cf-cf-preferences-js-git-90432a4a'...`

Note that I can 'git clone' the same package/project (cf-preferences-js), it's just npm install does not work for me. Other developer can run same npm install command, I believe issue is probably config on my laptop, but I have the same .gitconfig and .npmrc as he does.

#### How can the CLI team reproduce the problem?

Run: 
npm install git+https://acme.com/scm/cf/cf-preferences-js.git -save

error:
https://gist.github.com/walhow/5cf4ed4ce7c83101f9e446bb240d81f7

### supporting information:

 - `npm -v` prints: v4.2.3
 - `node -v` prints: v3.10.9
 - `npm config get registry` prints:

λ npm config get registry
https://registry.npmjs.org/

λ npm config list
; cli configs
user-agent = "npm/3.10.9 node/v4.2.3 win32 x64"

; userconfig C:\Users\user-xyz\.npmrc
https-proxy = "http://proxy.acme.com:80/"
prefix = "C:\\Users\\user-xyz\\.node_modules_global"
proxy = "http://proxy.acme.com:80/"

; node bin location = C:\Users\user-xyz\sw\Node\node.exe
; cwd = c:\dev\test1
; HOME = C:\Users\user-xyz
; "npm config ls -l" to show all defaults.

 - Windows, OS X/macOS, or Linux?: Windows
 - Network issues:
   - Geographic location where npm was run:
   - [ ] I use a proxy to connect to the npm registry.
   - [ x] I use a proxy to connect to the web.
   - [ x] I use a proxy when downloading Git repos.
   - [ ] I access the npm registry via a VPN
   - [ ] I don't use a proxy, but have limited or unreliable internet access.
 - Container:
   - [ ] I develop using Vagrant on Windows.
   - [ ] I develop using Vagrant on OS X or Linux.
   - [ ] I develop / deploy using Docker.
   - [ ] I deploy to a PaaS (Triton, Heroku).

**Comment by zkat** [comment]
2 things:

1. There seems to be some kind of newline in your clone. Any idea where it might've come from? That's kinda weird.
2. Can you try the command again, but instead of copy-pasting the same command you just ran, can you do `npm install -S <paste or type the URL of your repo here>`? There's sometimes weird things that happen when you paste funny characters by accident into a terminal.

**Comment by walhow** [comment]
Thanks, I tried again being careful to avoid weird characters, still get the 'Coning into bare repository' error.

**Comment by zkat** [comment]
@walhow can you clone that repository outside of npm? Does `git clone https://acme.com/scm/cf/cf-preferences-js.git` work at all, manually?

**Comment by walhow** [comment]
Yes, I can `git clone` it. Another team mate can `npm install` it, so I think it is something related to config or software on my machine, but my npmrc and gitconfig seem the same as his.

**Comment by zkat** [comment]
@walhow: Are you using a `GIT_PROXY` environment variable or such? npm pretty aggressively strips the environment. Additionally, is your teammate in your same office, with the same network, and using the same proxy?

**Comment by walhow** [comment]
zkat, thanks for the help! I'm punting on this for now, I think it is a configuration issue on my laptop, other people in the office can pull from the repo. We don't have admin rights on the machines, so it is tough to get these things fixed.

**Comment by joelwass** [comment]
Some wifi networks will muck with your ability to npm install. Try making sure you're on a secure network that isn't disabling certain types of network traffic.

**Comment by nicolelovecoding** [comment]
@walhow  did you solve this issue? I came into the same problem.

**Comment by walhow** [comment]
Sorry, no luck, but I think the problem was the proxy settings in my
npm.config

On Thu, Jul 20, 2017 at 12:54 PM, nicolelovecoding <notifications@github.com

**Comment by chenjienan** [comment]
Try to upgrade node version

**Comment by anjmittu** [comment]
I had this issue when running the command `npm install` using cygwin/babun terminal.  It was a very similar issue where my teammates could install from the private repository, but I would get an error. 

I solved this issue by running the command in the windows command prompt instead.

**Comment by walhow** [comment]
I switched to running npm commands in GitCMD, and no issues now.
